docs/source/signal_hooking.rst

```diff
@@ -0,0 +1,108 @@
+Signals
+=======
+
+libdebug supports hooking of signals in a similar way to syscalls. You can, in fact, execute a callback when a specific signal directed at the debugged process is intercepted by the tracer. \
+The following is the signature of the callback function:
+
+.. code-block:: python
+
+    def callback(d: ThreadContext, signal_number: int) -> None:
+
+along with the thread where the signal was intercepted from, the callback is also passed the signal number. \
+
+When registering a signal hook, you can either specify the signal number or the conventional signal name (e.g. 'SIGINT').
+
+.. code-block:: python
+
+    # Define hooks
+    def hook_SIGUSR1(t, signal_number):
+        t.signal = 0x0
+        print("Look mum, I'm hooking a signal")
+
+    def hook_SIGINT(t, signal_number):
+        print("Look mum, I'm hooking another signal")
+
+    # Register hooks
+    hook1 = d.hook_signal(10, callback=hook_SIGUSR1)
+    hook2 = d.hook_signal('SIGINT', callback=hook_SIGINT)
+
+    d.cont()
+
+    d.unhook_signal(hook1)
+
+Note: You cannot hook **SIGSTOP**, **SIGTRAP**, and **SIGKILL**.
+
+These signals are internally used by the ptrace and the debugger, or are enforced by the kernel to be passed directly to the child process without the possibility of being caught.
+
+Just like with syscalls, there can be at most one user-defined hook for each signal.
+
+If a new hook is defined for a signal that is already hooked or hijacked, the new hook will replace the old one, and a warning is printed.
+
+Signal Filtering
+----------------
+Instead of setting a callback on signals, you could want to filter which signals are not to be forwarded to the debugged process during execution.
+
+By default, all signals not related to the libdebug's internals are forwarded. For example, SIGSTOP is never passed to the process.
+
+.. code-block:: python
+    
+    d.signals_to_block = [10, 15, 'SIGINT', 3, 13]
+
+
+The same syntax will work in multithreaded applications, by setting `signals_to_block` on the desired thread context object. See :doc:`multithreading` for more information.
```

Comment:
Actually it is a list stored into the debugger, not in the ThreadContext. It is global. Good point for future improvements anyway :)